openwisp_controller/config/api/filters.py

```diff
@@ -0,0 +1,103 @@
+from django.utils.translation import gettext_lazy as _
+from django_filters import rest_framework as filters
+from swapper import load_model
+
+Template = load_model('config', 'Template')
+Vpn = load_model('config', 'Vpn')
+Device = load_model('config', 'Device')
+DeviceGroup = load_model('config', 'DeviceGroup')
+
+
+class BaseConfigAPIFilter(filters.FilterSet):
+    organization = filters.CharFilter(
+        field_name='organization', label=_('Organization')
+    )
+
+    def _set_valid_filterform_lables(self):
+        # When not filtering on a model field, an error message
+        # with the label "[invalid_name]" will be displayed in filter form.
+        # To avoid this error, we need to provide the label explicitly.
+        raise NotImplementedError
+
+
+class TemplateListFilter(BaseConfigAPIFilter):
+    class Meta:
+        model = Template
+        fields = {
+            'organization': ['exact'],
+            'backend': ['exact'],
+            'type': ['exact'],
+            'default': ['exact'],
+            'required': ['exact'],
+            'created': ['exact', 'gte', 'lt'],
+        }
+
+
+class VPNListFilter(BaseConfigAPIFilter):
+
+    subnet = filters.CharFilter(field_name='subnet', label=_('VPN Subnet'))
+
+    class Meta:
+        model = Vpn
+        fields = {
+            'backend': ['exact'],
+            'subnet': ['exact'],
+            'organization': ['exact'],
+        }
+
+
+class DeviceListFilter(BaseConfigAPIFilter):
+    config__templates = filters.CharFilter(
+        field_name='config__templates', label=_('Config template')
+    )
```

Comment:
I tried passing two UUIDs, comma separated, to see if it handled and the API returned an internal server error for invalid UUID. Please add a test for this and make sure the API returns 400 bad request.

Reply:
sure